service/raft/group0_state_machine.cc

```diff
@@ -83,12 +123,98 @@ future<> group0_state_id_handler::advertise_state_id(utils::UUID state_id) {
     return _gossiper.get().add_local_application_state(gms::application_state::GROUP0_TOMBSTONE_GC_TIME, gms::versioned_value::gc_clock(gc_time));
 }
 
+future<> group0_state_id_handler::on_change(gms::inet_address endpoint, const gms::application_state_map& states, gms::permit_id pid) {
+    return on_application_state_change(
+        endpoint,
+        states,
+        gms::application_state::GROUP0_TOMBSTONE_GC_TIME,
+        pid,
+        [this](gms::inet_address endpoint, const gms::versioned_value& value, gms::permit_id) {
+            const auto gc_time = gc_clock::from_time_t(std::stol(value.value()));
+
+            slogger.debug("Tombstone GC time received: {}, endpoint count: {}", gc_time, _gossiper.get().get_endpoints().size());
+
+            _gc_times[endpoint] = gc_time;
+
+            auto* const group0_server = _server_accessor.get_server();
+            if (!group0_server) {
+                slogger.debug("Group0 server not found");
+                return make_ready_future<>();
+            }
+
+            const auto& group0_members = group0_server->get_configuration().current;
+            if (group0_members.empty()) {
+                slogger.debug("Group0 has no members");
```

Comment:
These log messages are using the general group 0 state machine logger so we will see something like
`group0_state_machine - Group 0 has no members`

missing the important info that this is printed in the context of the tombstone GC handler currently handling a notification.

Reply:
done, using a separate log - in fact, moved the whole `group0_state_id_handler` to a separate file (header/src) as it is now being created and used and 2 different places